### PR TITLE
th/keyfile-various

### PR DESCRIPTION
**Pull request by thom311**
Some refactoring of keyfile.

These changes are in preparation for the multiple-keyfile-directories work. Here proposed to merge early, to split up the work into smaller pieces.

Please review.

### CODE REVIEW

File: src/devices/tests/test-acd.c [hunk 1]
```diff
@@ -31,6 +33,46 @@
...
+
+#define _skip_acd_test() \
+	({ \
+		gboolean _skip = _skip_acd_test_check (); \
+		\
+		if (_skip) \
+			g_test_skip ("Cannot create NAcd. Running under valgind?"); \
+		_skip; \
+	})
+
```
Comment (bengal) [inline src/devices/tests/test-acd.c]: Wouldn't it be simpler to skip the test from makefile/meson.build when valgrind is enabled?

Comment (thom311) [reply src/devices/tests/test-acd.c]: no. The configure option is mostly irrelevant and only affects the default. What matters is whether you enable valgrind when running the test:
```
   NMTST_USE_VALGRIND=1 make -j 5 check
   NMTST_USE_VALGRIND=1 ./tools/run-nm-test.sh ./src/devices/tests/test-acd
   ./tools/run-nm-test.sh -v ./src/devices/tests/test-acd
```
vs.
```
   NMTST_USE_VALGRIND=0 make -j 5 check
   NMTST_USE_VALGRIND=0 ./tools/run-nm-test.sh ./src/devices/tests/test-acd
   ./tools/run-nm-test.sh -V ./src/devices/tests/test-acd
```

Comment (bengal) [reply src/devices/tests/test-acd.c]: Understood, thanks.

**Comment by thom311** [comment]
merged

**Pull request closed without merging**